files/usr/share/cinnamon/applets/menu@cinnamon.org/applet.js

```diff
@@ -1412,39 +1225,48 @@ class CinnamonMenuApplet extends Applet.TextIconApplet {
         }
     }
 
-    _recentMenuOpenStateChanged(recentContextMenu) {
-        if (recentContextMenu.isOpen) {
-            this._activeContextMenuParent = recentContextMenu.sourceActor._delegate;
-            this._scrollToButton(recentContextMenu);
+    _contextMenuOpenStateChanged(menu) {
+        if (menu.isOpen) {
+            this._activeContextMenuParent = menu.sourceActor._delegate;
+            this._scrollToButton(menu);
         } else {
             this._activeContextMenuItem = null;
             this._activeContextMenuParent = null;
-            for (let item in this._recentButtons) {
-                if (this._recentButtons[item].menu) {
-                    this._recentButtons[item].menu = null;
-                }
-            }
+            menu.sourceActor = null;
         }
     }
 
-    createRecentContextMenu(actor) {
-        let menu = new PopupMenu.PopupSubMenu(actor);
-        menu.actor.set_style_class_name('menu-context-menu');
-        menu.connect('open-state-changed', Lang.bind(this, this._recentMenuOpenStateChanged));
-        this.recentContextMenu = menu;
+    toggleContextMenu(button) {
+        if (!button.withMenu)
+            return;
+
+        if (!this.contextMenu) {
+            let menu = new PopupMenu.PopupSubMenu(null); // hack: creating without actor
+            menu.actor.set_style_class_name('menu-context-menu');
+            menu.connect('open-state-changed', Lang.bind(this, this._contextMenuOpenStateChanged));
+            this.contextMenu = menu;
+            this.applicationsBox.add_actor(menu.actor);
+        }
```

Comment:
Hello, @itzexor.

A little suggestion to improve the context menu logic.

#### Current behaviour

Attempting to open the context menu of any button with right click while another context menu is opened requires two clicks on a button. One to close the previously opened menu, and another to open the desired menu.

#### Proposed behaviour

Attempting to open the context menu of any button while another context menu is opened should close the opened menu and at the same time open the context menu of the clicked button.

I managed to achieve the proposed behaviour with the following changes:

```suggestion
        }

        if (this.contextMenu.sourceActor !== button.actor &&
            this.contextMenu.isOpen) {
            this.contextMenu.close();
        }

```